src/hooks/useStartWatchingButton.ts

```diff
@@ -0,0 +1,44 @@
+import { useCallback, useMemo } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useHistory } from 'react-router';
+
+import { addQueryParam } from '../utils/history';
+
+export type UseStartWatchingLabel = (
+  isEntitled: boolean,
+  hasMediaOffers: boolean,
+  isLoggedIn: boolean,
+  videoProgress?: number,
+  videoUrl?: string | null,
+) => {
+  startWatchingLabel: string;
+  handleStartWatchingClick: () => void;
+};
+
+const useStartWatchingButton: UseStartWatchingLabel = (isEntitled, hasMediaOffers, isLoggedIn, videoProgress, videoUrl) => {
```

Comment:
Couldn't you have reused the click handler in a component too?  And then you have the benefit of encapsulating the button logic in 1 place and not needing to pass properties.

Reply:
I've created a StartWatchingButton container which now contains all the logic (not a hook for now).